src/validation.h

```diff
@@ -343,12 +343,17 @@ bool HasValidProofOfWork(const std::vector<CBlockHeader>& headers, const Consens
 /** Return the sum of the work on a given set of headers */
 arith_uint256 CalculateHeadersWork(const std::vector<CBlockHeader>& headers);
 
+enum class VerifyDBResult {
+    SUCCESS,
+    ERROR_CORRUPTED_BLOCK_DB,
+};
+
 /** RAII wrapper for VerifyDB: Verify consistency of the block and coin databases */
 class CVerifyDB {
 public:
     CVerifyDB();
     ~CVerifyDB();
-    bool VerifyDB(
+    VerifyDBResult VerifyDB(
```

Comment:
e7dfc743406bfa0cac94c640dd5eaeac4364c990: should be nodiscard?

Reply:
done